src/sonic-yang-models/yang-models/sonic-bgp-common.yang

```diff
@@ -0,0 +1,540 @@
+                    the neighbor or group";
+            }
+        }
+        description
+            "Type to describe the add paths advertisement method.";
+    }
+
+    typedef bgp_community_type {
+        type enumeration {
+            enum standard {
+                description "Standard communities";
+            }
+
+            enum extended {
+                description "Extended communities";
+            }
+
+            enum both {
+                description "Both standard and extended communities";
+            }
+
+            enum large {
+                description "Large communities";
+            }
+
+            enum all {
+                description "Standard, Extended and Large communities";
+            }
+
+            enum none {
+                description "No community attribute";
+            }
+        }
+        description
+            "type describing variations of community attributes";
+    }
+
+    typedef sonic_bgp_direction {
+        type enumeration {
+            enum in {
+                description
+                    "Refers to information received from the BGP peer";
+            }
+            enum out {
+                description
+                    "Refers to information advertised to the BGP peer";
+            }
+        }
+        description
+            "Type to describe the direction";
+    }
+
+    typedef sonic_bgp_orf {
+        type enumeration {
+            enum send {
+                description
+                    "Capability to receive the outbound route filtering from this neighbor";
+            }
+            enum receive {
+                description
+                    "Capability to send the outbound route filtering to this neighbor";
+            }
+            enum both {
+                description
+                    "Capability to send and receive the outbound route filtering to/from this neighbor";
+            }
+        }
+        description
+            "Type to describe the BGP ORF(Outbound route filtering) capability";
+    }
+
+    grouping sonic-bgp-cmn {
+
+        leaf local_asn {
+            type uint32 {
+                range "1..4294967295";
+            }
+            description "Local AS number";
+        }
+
+        leaf name {
+            type string;
+            description "Peer description";
+        }
+
+        leaf asn {
+            type uint32 {
+                range "1..4294967295";
+            }
+            description "Peer AS number";
+        }
+
+        leaf peer_type {
+            type bgp_peer_type;
+            description "BGP peer type internal/external";
+        }
+
+        leaf ebgp_multihop {
+            type boolean;
+            description "EBGP Multihop enabled";
+        }
+
+        leaf ebgp_multihop_ttl {
+            type uint8 {
+                range "1..255";
+            }
+            description "EBGP Multihop TTL";
+        }
+
+        leaf auth_password {
+            type string;
```

Comment:
There should be a flag "encrypted" to indicate if the password is in encrypted format

Reply:
It is always encrypted as we store only encrypted password in the DB.